src/fireedge/src/client/components/Forms/ServiceTemplate/CreateForm/Steps/Extra/customAttributes/schema.js

```diff
@@ -0,0 +1,181 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2023, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+import { string, boolean, number } from 'yup'
+import { getObjectSchemaFromFields, arrayToOptions } from 'client/utils'
+import { INPUT_TYPES, T } from 'client/constants'
+
+const getTypeProp = (type) => {
+  switch (type) {
+    case CA_TYPES.boolean:
+      return INPUT_TYPES.SWITCH
+    case CA_TYPES.text:
+    case CA_TYPES.text64:
+    case CA_TYPES.number:
+    case CA_TYPES.numberfloat:
+      return INPUT_TYPES.TEXT
+    default:
+      return INPUT_TYPES.TEXT
+  }
+}
+
+const getFieldProps = (type) => {
+  switch (type) {
+    case CA_TYPES.text:
+    case CA_TYPES.text64:
+      return { type: 'text' }
+    case CA_TYPES.number:
+    case CA_TYPES.numberfloat:
+      return { type: 'number' }
+    default:
+      return {}
+  }
+}
+
+// Define the CA types
+const CA_TYPES = {
+  text64: 'text64',
+  password: 'password',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. This poses a serious security risk as it may allow unauthorized access to sensitive data or systems. Possible remediation approaches are:
	 1. Manage secret data using environment variables:
		 a) Identify the secret data and store that data in environment variables on your local and production server.
		 b) Replace the secrets with references to environment variables.
	 2. Use AWS Secrets Manager to store and access secrets:
		 a) Visit Secrets Manager on the AWS Management Console.
		 b) Choose the secret type on the console page and follow the instructions.
		 c) Use the suggested code samples to retrieve the secret.

To learn more about password management best practices, see [OWASP password storage cheat sheet](https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet).